rust/src/common.rs

```diff
@@ -72,3 +72,20 @@ pub unsafe extern "C" fn rs_cstring_free(s: *mut c_char) {
     }
     drop(CString::from_raw(s));
 }
+
+/// Take ownership of `obj`, box it, and returns it's raw pointer value
+pub fn rust_to_c_boxed<T>(obj: T) -> *mut T {
+    Box::into_raw(Box::new(obj))
+}
+
+/// Free a Box'd value allocated by Rust (for ex. using `rust_to_c_boxed`)
+///
+/// # Safety
+///
+/// `ptr` must be a pointer allocated via the Box interface
+pub unsafe extern "C" fn rs_free_boxed<T>(ptr: *mut T) {
+    if ptr.is_null() {
+        return;
+    }
+    drop(Box::from_raw(ptr));
+}
```

Comment:
Can you remove these functions and the usage of them for now. I think some generalization may be needed in this area, just not confident on how this should look yet, so I don't want to introduce something like this yet.

Reply:
No problem. I'll remove the last commit.